### PR TITLE
Add after join validation and JSON rendering

### PR DESCRIPTION
**Pull request by carsten-luckmann**

### CODE REVIEW

File: pipeline/customer/customer.py [hunk 1]
```diff
@@ -11,6 +11,69 @@
...
+    Orders = 'orders'
+
+    Errors = 'errors'
+
+
+class Validate(beam.DoFn):
+
+    def process(self, element, *args, **kwargs):
+        customer_id, entry = element
+        details = list(entry.get(Field.Detail, []))
```
Comment (Xennis) [inline pipeline/customer/customer.py:31]: Add a comment why you cast here (i.e. we get only an iterable but want to have a list)

File: pipeline/customer/customer_test.py [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+                        'id': 3949,
+                        'total_price': Decimal('702.88'),
+                    }],
+                order.Field.Min: Decimal('373.02'),
+                order.Field.Max: Decimal('702.88'),
+                order.Field.Sum: Decimal('1075.90'),
+                order.Field.Avg: Decimal('537.95'),
+                order.Field.Error: [],
+            },
+            customer.Field.Errors: [u'no customer details'],
```
Comment (Xennis) [inline pipeline/customer/customer_test.py]: The `u'` (unicode) is not needed because of the `from __future__ import unicode_literals`.

**Review by Xennis — COMMENTED** [review]
Nothing to add. See first code review.

File: pipeline/customer/customer.py (hunk 1 above)
Comment (carsten-luckmann) [reply pipeline/customer/customer.py:31]: done

File: pipeline/customer/customer_test.py (hunk 2 above)
Comment (carsten-luckmann) [reply pipeline/customer/customer_test.py]: done

**Review by Xennis — APPROVED** [review]

**Pull request merged**